File: CS/CustomEditorWinApp3/RepositoryItemCustomLookUpEdit.cs
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace CustomEditor
{
	class RepositoryItemCustomLookUpEdit : RepositoryItemLookUpEdit
	{
		internal const string EditorName = "CustomLookUpEdit";

		static RepositoryItemCustomLookUpEdit()
		{
			Register();
		}
		public RepositoryItemCustomLookUpEdit()
		{
		}

		public static void Register()
		{
			EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName,
			  typeof(CustomLookUpEdit), typeof(RepositoryItemCustomLookUpEdit),
				typeof(ButtonEditViewInfo), new ButtonEditPainter(), true));
		}

		public override string EditorTypeName
		{
			get
			{
				return EditorName;
			}
		}

		protected internal new bool IsNullInputAllowed
		{
			get
			{
				return base.IsNullInputAllowed;
			}
		}
	}
}

File: VB/CustomEditorWinApp3/RepositoryItemCustomLookUpEdit.vb
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace CustomEditor

    Friend Class RepositoryItemCustomLookUpEdit
        Inherits RepositoryItemLookUpEdit

        Friend Const EditorName As String = "CustomLookUpEdit"

        Shared Sub New()
            Call Register()
        End Sub

        Public Sub New()
        End Sub

        Public Shared Sub Register()
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(CustomLookUpEdit), GetType(RepositoryItemCustomLookUpEdit), GetType(ButtonEditViewInfo), New ButtonEditPainter(), True))
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return EditorName
            End Get
        End Property

        Protected Friend Overloads ReadOnly Property IsNullInputAllowed As Boolean
            Get
                Return MyBase.IsNullInputAllowed
            End Get
        End Property
    End Class
End Namespace
